Add more HTTP status codes (#12601)
mdn/content: files/en-us/web/http/status/207/index.md

---
title: 207 Multi Status
slug: Web/HTTP/Status/207
tags:
  - HTTP
  - Status code
  - Partial Success
spec-urls: https://www.rfc-editor.org/rfc/rfc4918#section-11.1
---

{{HTTPSidebar}}

> The HTTP **`207 Multi Status`** response code indicates that there might be mixture of responses. We use {{HTTPStatus(204)}} if all the operations succededs or else if all fails we can use {{HTTPStatus(403)}}. But if half of them fails and half succedds we use Multi Status in such conditions.

Reviewer: @teoli2003 I fixed the typo and English  here, but I'm not sure what "multi status" means below.

```suggestion
The HTTP **`207 Multi Status`** response code indicates that there might be mixture of responses.
We use {{HTTPStatus(204)}} if all the operations succeed and {{HTTPStatus(403)}} if all operations fail.
If some operations succeed and others fail then we use Multi Status.
```

Reviewer 2: I have revamped the page. Example of return status are useless here (it was far from being exhaustive anyway)

---
title: 207 Multi Status
slug: Web/HTTP/Status/207
tags:
  - HTTP
  - Status code
  - Partial Success
spec-urls: https://www.rfc-editor.org/rfc/rfc4918#section-11.1
---

{{HTTPSidebar}}

The HTTP **`207 Multi Status`** response code indicates that there might be mixture of responses.

We use {{HTTPStatus(204)}} if all the operations succeed and {{HTTPStatus(403)}} if all operations fail.
`207 Multi Status` is returned if some operations succeed and others fail.

> If you perform a destructive operation like {{HTTPMethod("POST")}}, {{HTTPMethod("DELETE")}} etc. against more than one resource and the operations against each individual resource does not share a common outcome then you can go for 207.

Reviewer: I thought each operation was atomic so if you deleted a resource it would succeed or fail in entirety. I assume this is not the case. Is this wording correct?

```suggestion
If you perform a destructive operation like {{HTTPMethod("POST")}}, {{HTTPMethod("DELETE")}} etc. against more than one resource and the operations against each individual resource do not not share a common outcome then the response can be `207`.
```
